### PR TITLE
create unit tests for argParse class

### LINKED ISSUE #95: create unit tests for argParse class

### PR DESCRIPTION
**Pull request by barbosamaatheus**
I added unit tests for the **argsParser** class and achieved coverage of: 100% of the class, 88% of the methods and 92% of the code lines.

In addition, I made changes to the **argsParser** class to correct the following problems:
1. When an **outputFile** ending with '/' is sent the error is returned: 
```
No such property: dir for class: arguments.ArgsParser
groovy.lang.MissingPropertyException: No such property: dir for class: arguments.ArgsParser
```

2. Return `true` when sending an invalid date to the`validDate` method
3. upgrade to junit 5 while maintaining junit4 compatibility
4. add validation for "Invalid argument passed"

Resolves #95

### CODE REVIEW

File: src/test/unit/ArgsParserTest.groovy [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        Arguments resultArgs = argsParser.parse(args)
+
+        Arguments argsDefault = new Arguments()
+        argsDefault.setHelp()
+
+        assertEquals(argsDefault.getProperties(), resultArgs.getProperties())
+    }
+
+    @Test
+    public void invalidArgument() {
```
Comment (rafaelmotaalves) [inline src/test/unit/ArgsParserTest.groovy:80]: Is there any way we could improve this behavior? throwing an error that better represents this scenario, something like: "Invalid argument passed"

File: src/test/unit/ArgsParserTest.groovy [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        String[] args = ['--invalid', 'invalid','./projects.csv']
+
+        try {
+            argsParser.parse(args)
+        } catch (InvalidArgsException e) {
```
Comment (rafaelmotaalves) [inline src/test/unit/ArgsParserTest.groovy:85]: Using this way of asserting exceptions, if no exception is thrown the test will not fail. Consider using one of the asserting tools described here: https://www.baeldung.com/junit-assert-exception

Comment (barbosamaatheus) [reply src/test/unit/ArgsParserTest.groovy:85]: I saw this post before implementing it. However, as we use junit 4, the first approach is not possible to catch the message thrown in the exception. The second approach uses `ExpectedException.none();` where the `none()` method is deprecated and I prefer not to use it.

File: src/test/unit/ArgsParserTest.groovy (hunk 1 above)
Comment (barbosamaatheus) [reply src/test/unit/ArgsParserTest.groovy:80]: I believe that if we place a new validation verifying if the arguments passed are different from those expected, we would throw an exception with a personalized message. Should I do this?

File: src/test/unit/ArgsParserTest.groovy (hunk 2 above)
Comment (rafaelmotaalves) [reply src/test/unit/ArgsParserTest.groovy:85]: ok, another alternative, is to use assert to guarantee that the catch is executed, doing some like:
```groovy
try {
   argsParser.parse(args)
   assertEquals(true, false, "InvalidArgsException was not thrown");
} catch (InvalidArgsException e) {
...
}
```
but that is kind of a hack

Comment (rafaelmotaalves) [reply src/test/unit/ArgsParserTest.groovy:85]: if you want we can also try to upgrade JUnit' version

File: src/test/unit/ArgsParserTest.groovy (hunk 1 above)
Comment (rafaelmotaalves) [reply src/test/unit/ArgsParserTest.groovy:80]: yes, I think this would be a good fix

File: src/test/unit/ArgsParserTest.groovy (hunk 2 above)
Comment (barbosamaatheus) [reply src/test/unit/ArgsParserTest.groovy:85]: If we use:
```java
try {
   argsParser.parse(args)
   fail("expected exception was not occured.");
} catch (InvalidArgsException e) {
   assertEquals(e.getMessage(), 'message');
}
```
Looks better to you?

I found here: https://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests

Comment (rafaelmotaalves) [reply src/test/unit/ArgsParserTest.groovy:85]: yes, its better, but if could use JUnit5 probably would be better

Comment (barbosamaatheus) [reply src/test/unit/ArgsParserTest.groovy:85]: Can you tell if Junit 5 is compatible with all Junit4 tests?

Comment (rafaelmotaalves) [reply src/test/unit/ArgsParserTest.groovy:85]: i don't know, probably is because we only use basic Junit features, but can test if it really is

Comment (barbosamaatheus) [reply src/test/unit/ArgsParserTest.groovy:85]: I found this article about migrating from junit4 to junit5: https://www.baeldung.com/junit-5-migration

Can you see and verify the compatibility on our project?

File: src/test/unit/ArgsParserTest.groovy (hunk 1 above)
Comment (barbosamaatheus) [reply src/test/unit/ArgsParserTest.groovy]: correction made on: https://github.com/spgroup/miningframework/pull/96/commits/410c4d2e9bc77ddb9fbf22373edc2e6c07c02cd6

File: src/test/unit/ArgsParserTest.groovy (hunk 2 above)
Comment (barbosamaatheus) [reply src/test/unit/ArgsParserTest.groovy]: upgrade to junit 5 while maintaining junit4 compatibility: https://github.com/spgroup/miningframework/pull/96/commits/445803eb1e7af0f98f37249ef923ff2e642f7d31
re-write tests in junit 5 improving exception capture: https://github.com/spgroup/miningframework/pull/96/commits/078c96843c53ba15aa82b15f0d028615b5926dd4

**Pull request merged**